File: Strategy Pattern/DuckSim_py/Ente/Moorente.py
from typing import Type

from IFlugverhalten import IFlugverhalten
from IQuakverhalten import IQuakverhalten
from Ente import AEnte


class Moorente(AEnte.AEnte):
    def __init__(self, fv: Type[IFlugverhalten], qv: Type[IQuakverhalten]):
        super(Moorente, self).__init__(fv, qv)

    def anzeigen(self) -> None:
        super().anzeigen()
        print("Moorente")

File: Strategy/DuckSim_cpp/Moorente.cpp
#include "Moorente.h"

Moorente::Moorente(IQuakverhalten* qv, IFlugverhalten* fv)
	:Ente(qv, fv)
{
}

void Moorente::anzeigen()
{
	Ente::anzeigen();
	std::cout << "Moorente\n";
}
